src/FakeItEasy.Analyzer/SymbolExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.CodeAnalysis;
+
+    public static class SymbolExtensions
+    {
+        public static string GetFullName(this INamedTypeSymbol type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            var stack = new Stack<string>();
+            stack.Push(type.Name);
+            var tmp = type.ContainingType;
+            while (tmp != null)
+            {
+                type = tmp;
+                stack.Push(tmp.Name);
+                tmp = tmp.ContainingType;
+            }
+
+            var ns = type.ContainingNamespace;
+            while (!string.IsNullOrEmpty(ns?.Name))
+            {
+                stack.Push(ns.Name);
+                ns = ns.ContainingNamespace;
+            }
+
+            return string.Join(".", stack);
+        }
+    }
+}

```

Comment:
Oh! Sorry :worried: 

BTW - I had a play to see if I could boil the method a bit and this is what I came up with:

```C#
        public static string GetFullName(this ISymbol symbol)
        {
            if (symbol == null)
            {
                throw new ArgumentNullException(nameof(symbol));
            }

            var names = new Stack<string>();
            names.Push(symbol.Name);
            while ((symbol = symbol.ContainingType) != null)
            {
                names.Push(symbol.Name);
            }

            while ((symbol = symbol.ContainingNamespace) != null)
            {
                names.Push(symbol.Name);
            }

            return string.Join(".", names);
        }
```